Makefile

```diff
@@ -24,10 +24,13 @@ bundle-docs-clean:
 
 bundle-docs: bundle-docs-clean
 	cd docs && \
-	make html && \
+	make html SPHINXOPTS="-W --keep-going" && \
```

Comment:
@ahangsu Thanks for the earlier test to prove the highlighted line _can_ cause the build to fail (https://github.com/algorand/pyteal/runs/7568763886?check_suite_focus=true).

I'm wondering your thoughts on including `-n` to check for broken links?

* I see at least one commit where you tried it (https://github.com/algorand/pyteal/commit/35ab49b0e5cb53103bf9e9d22980ee4935cdf80f).  Though it's unclear why it was reverted.
* I've not tried using the option though it _seems_ like a prudent check.  If you haven't observed issues with `-n`, I'm up for trying it.

Reply:
Alright - I found a related build with `-n` enabled:  https://github.com/algorand/pyteal/runs/7568763886?check_suite_focus=true.

Seems it generates a number of warnings.  I briefly reviewed and started googling though I don't know off-hand how to resolve.

Without further inspection, it's unclear to me how these types of warnings manifest themselves in the docs.  I'm trying to gauge if there's specific warnings here that make `-n` worth retaining.

It's also possible we agree that adding `-n` is an exercise for another day.  Since we're here, I'd appreciate a brief look on your side to better understand the situation.

Reply:
I am collecting the `nitpicky-mode` activated by option `-n`, but not too much information is collected, and I am not too sure of if it is out of scope of the ticket, but I am happy to keep exploring.

AFAIK, `nitpicky-mode`: warn about all references where the target cannot be found, which seems to be a good practice. But this would enforce us to rewrite a bunch of docstrings, e.g.,:
- we sometimes use `(optional)` for args, while this might cause warning in Sphinx nitpicky mode.

I somewhat think adding `-n` is an exercise left to be done some other day, for it would expand the scope of work at this point...

Reply:
I added a few fixes and configs for nitpicky mode, and resolved most of the warnings in this PR. Take a pass if you are in free loop.

Reply:
after the discussion we decided to close this thread, for we will revisit `-n` nitpick mode in the future. At this moment, we still cannot specify a class un-exported to be not documented or referenced in Sphinx.